src/path.rs

```diff
@@ -35,15 +35,14 @@ where
 {
     pub fn new(namespace: &[u8], keys: &[&[u8]]) -> Self {
         let l = keys.len();
-        // FIXME: make this more efficient
-        let storage_key = nested_namespaces_with_key(
-            &[namespace],
-            &keys[0..l - 1]
-                .iter()
-                .map(|k| Key::Ref(k))
-                .collect::<Vec<Key>>(),
-            keys[l - 1],
-        );
+
+        // Combine namespace and all but last keys.
+        // This is a single vector allocation with references as elements.
+        let mut combined: Vec<&[u8]> = Vec::with_capacity(1 + keys.len() - 1);
+        combined.push(namespace);
+        combined.extend(keys[0..l - 1].iter());
+        debug_assert_eq!(combined.capacity(), combined.len()); // ensure no reallocation needed
```

Comment:
Technically, `Vec::with_capacity` is [allowed to allocate more than the given amount](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity). I don't know if this ever happens in practice, but I would prefer to compare capacity before and after the pushes instead.

Reply:
Oh, good point. I was not aware of that.

If we have to check before vs. after, we can also compare the pointers which directly checks the reallocation. But this might be a bit overkill.

Reply:
What about this? [e375e2a](https://github.com/CosmWasm/cw-storage-plus/pull/64/commits/e375e2abc13ea0ba5a8c4d6b9d3f9dfb3789f6c4)

Reply:
Yes, that works :+1: